### ISSUE TITLE
Rogue edges

### ISSUE THREAD

**Issue by samzmann** [Jonothy/contours#1]
We need to join edges of blobs that are incomplete. For now I have been able to draw edges on x=0 and y=0 axis. For some reason the same logic does not work on x=1 and y=1.
However, even on x=0 and y=0 the solution is not good because it does not take into account other incomplete blobs within that blobs.
Nest we should look into joining incomplete blobs that go over a corner, still checking for embedded incomplete blobs.

So to recap, the case are:

- blob with its incomplete edges on x=0, y=0, x=1 or y=1
- blob with incomplete edges on x=0, y=0, x=1 or y=1, and with another incomplete blob within it
- blob with incomplete edges on different axis (corner case)
- blob with incomplete edges on different axis and with another incomplete bob within it
